synapse/storage/databases/main/event_federation.py

```diff
@@ -135,7 +145,7 @@ async def get_auth_chain_ids(
 
         # Check if we have indexed the room so we can use the chain cover
         # algorithm.
-        room = await self.get_room(room_id)
+        room = await self.get_room(room_id)  # type: ignore[attr-defined]
```

Comment:
This is defined on the `RoomWorkerStore`. You could try making this class inherit from `RoomWorkerStore`, but... I'm not sure if that'll work. Our DataStore classes are a mess.

Reply:
I have tried different inheritance orders. I got always more mypy errors in different other files.